src/mlpack/methods/linear_svm/linear_svm_impl.hpp

```diff
@@ -48,10 +48,29 @@ LinearSVM<MatType>::LinearSVM(
     delta(delta),
     fitIntercept(fitIntercept)
 {
+  if (numClasses <= 1)
+  {
+    std::string error = "LinearSVM dataset has 0 number of classes!";
```

Comment:
Actually you could just put the string error directly into the constructor for `std::invalid_argument`. :)

In addition, it's valid to create a `LinearSVM` object without training it, so this check should only be done in `Train()`.  Maybe it is better to wait to initialize the weights until `Train()` is called too.

Reply:
Updated.